src/main/scala/org/broadinstitute/dsde/workbench/avram/api/AvramServlet.scala

```diff
@@ -0,0 +1,67 @@
+package org.broadinstitute.dsde.workbench.avram.api
+
+
+import scala.concurrent.duration.Duration
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+
+import org.broadinstitute.dsde.workbench.avram.Avram
+import org.broadinstitute.dsde.workbench.avram.dataaccess.SamUserInfoResponse
+import org.broadinstitute.dsde.workbench.avram.model.AvramException
+import org.broadinstitute.dsde.workbench.avram.util.ErrorResponse
+
+import scala.concurrent.Await
+import scala.util.{Failure, Success}
+//import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+trait AvramServlet {
+
+  implicit val executionContext = scala.concurrent.ExecutionContext.Implicits.global
+
+  private val samDao = Avram.samDao
+  private val bearerPattern = """(?i)bearer (.*)""".r
+
+  def registerOnComplete[T]
+  (future: Future[T], request: HttpServletRequest, response: HttpServletResponse)
+  (f: T => Unit): Future[T] = {
+    future onComplete {
+      case Success(result) => f(result)
+      case Failure(e: AvramException) => {
+        response.setStatus(e.status)
+        response.getWriter.write(e.message)
+      }
+      case Failure(e) => response.getWriter.write(e.getMessage)
+    }
+    future
+  }
+
+
+  def handleAuthenticatedRequest[T]
+  (request: HttpServletRequest, response: HttpServletResponse)
+  (f: SamUserInfoResponse => Future[Either[AvramException, T]]): T = {
+    val thing = for {
+      userInfo <- extractUserInfo(request)
+      result <- f(userInfo)
```

Comment:
This got a little bit jumbled so that `f()` is being invoked twice. The result of this invocation is being thrown away.